plugins/logs/middleware.go

```diff
@@ -187,15 +200,77 @@ func (l *Logs) recordResponse(request *Request, w *httptest.ResponseRecorder, re
 		}
 	}
 	if *reqCategory == category.ReactiveSearch {
-		var resBody ResponseBodyRS
-		err := json.Unmarshal(responseBody, &resBody)
+		// Read request body from context
+		rsRequestBody, err := request.FromContext(ctx)
+		// ignore error to record > 500 status code logs
 		if err != nil {
-			log.Errorln(logTag, "error encountered while parsing the response: ", err)
+			log.Errorln(logTag, "error encountered while reading request body:", err)
 		}
-		// ignore error to record error logs
-		if err == nil {
-			rec.Response.Took = &resBody.Settings.Took
+		marshalled, err := json.Marshal(rsRequestBody)
```

Comment:
Conversion to JSON may not be needed. https://stackoverflow.com/a/49007587/1221677

Reply:
@siddharthlatest It doesn't work https://play.golang.org/p/dBjzZLVn9WW